File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (needle.empty())
            return 0;
        
        for (int i = 0; i <= (int)haystack.length() - (int)needle.length(); ++i) {
            if (haystack.substr(i, needle.length()) == needle)
                return i;
        }

        return -1;
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle == "":
            return 0
        
        for i in range(len(haystack) + 1 - len(needle)):
            # does ever character of the needle match with a substring in the haystack
            for j in range(len(needle)):
                # i is the starting index, j acts as the increment
                if haystack[i+j] != needle[j]:
                    break
                if j == len(needle)-1:
                    return i
        
        return -1